File: app/src/main/java/com/ongouser/Home/activity/DeliveryOptions3Activity.java
package com.ongouser.Home.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.app.Dialog;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import com.ongouser.Home.DeliveryOptions3Adapter;
import com.ongouser.Login.LoginActivity;
import com.ongouser.R;

public class DeliveryOptions3Activity extends AppCompatActivity {
Context mContext;
Button btnConfirm;
Dialog dialog;
    RecyclerView recyclerview;
    DeliveryOptions3Adapter deliveryOptions3Adapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_delivery_options3);
        mContext=this;
        btnConfirm=findViewById(R.id.btnConfirm);
        btnConfirm.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                showDailog();
            }
        });
        ImageView ivBack=findViewById(R.id.ivBack);
        ivBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        recyclerview=findViewById(R.id.recyclerview);
        deliveryOptions3Adapter = new DeliveryOptions3Adapter(mContext);
        recyclerview.setLayoutManager(new LinearLayoutManager(mContext));
        recyclerview.setAdapter(deliveryOptions3Adapter) ;
    }
    public void showDailog(){

        dialog = new Dialog(mContext);
        dialog.getWindow().setBackgroundDrawableResource(android.R.color.transparent);

        dialog.setContentView(R.layout.alert_delivery3);
        dialog.setCancelable(true);

        Button btnOk= dialog.findViewById(R.id.btnOk);
        btnOk.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mContext.startActivity(new Intent(mContext, LoginActivity.class));
                dialog.dismiss();
            }
        });
        dialog.show();
    }
}

File: app/src/main/java/com/ongouser/home/activity/DeliveryOptions3Activity.kt
package com.ongouser.home.activity

import android.app.Dialog
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.ImageView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.ongouser.Login.LoginActivity
import com.ongouser.R
import com.ongouser.base.BaseActivity
import com.ongouser.Adapter.DeliveryOptions3Adapter

class DeliveryOptions3Activity : BaseActivity() {
    lateinit var mContext: Context
    lateinit var btnConfirm: Button
    lateinit var dialog: Dialog
    lateinit var recyclerview: RecyclerView
    lateinit var deliveryOptions3Adapter: DeliveryOptions3Adapter
    override fun getContentId(): Int {
        return R.layout.activity_delivery_options3
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mContext = this
        btnConfirm = findViewById(R.id.btnConfirm)
        btnConfirm.setOnClickListener(View.OnClickListener { showDailog() })
        val ivBack = findViewById<ImageView>(R.id.ivBack)
        ivBack.setOnClickListener { finish() }
        recyclerview = findViewById(R.id.recyclerview)
        deliveryOptions3Adapter = DeliveryOptions3Adapter(mContext)
        recyclerview.setLayoutManager(LinearLayoutManager(mContext))
        recyclerview.setAdapter(deliveryOptions3Adapter)
    }

    fun showDailog() {
        dialog = Dialog(mContext!!)
        dialog!!.window!!.setBackgroundDrawableResource(android.R.color.transparent)
        dialog!!.setContentView(R.layout.alert_delivery3)
        dialog!!.setCancelable(true)
        val btnOk = dialog!!.findViewById<Button>(R.id.btnOk)
        btnOk.setOnClickListener {
            mContext!!.startActivity(Intent(mContext, LoginActivity::class.java))
            dialog!!.dismiss()
        }
        dialog!!.show()
    }
}